mods/default/torch.lua

```diff
@@ -3,152 +3,116 @@
+		if node_def and node_def.on_rightclick and
 			not (placer and placer:is_player() and
 			placer:get_player_control().sneak) then
-			return def.on_rightclick(under, node, placer, itemstack,
+			return node_def.on_rightclick(under, node, placer, itemstack,
 				pointed_thing) or itemstack
 		end
-
 		local above = pointed_thing.above
 		local wdir = minetest.dir_to_wallmounted(vector.subtract(under, above))
-		local fakestack = itemstack
 		if wdir == 0 then
-			fakestack:set_name("default:torch_ceiling")
+			itemstack:set_name(name .. "_ceiling")
 		elseif wdir == 1 then
-			fakestack:set_name("default:torch")
+			itemstack:set_name(name)
 		else
-			fakestack:set_name("default:torch_wall")
+			itemstack:set_name(name .. "_wall")
 		end
+		itemstack = minetest.item_place(itemstack, placer, pointed_thing, wdir)
+		itemstack:set_name(name)
+		return itemstack
+	end
 
-		itemstack = minetest.item_place(fakestack, placer, pointed_thing, wdir)
-		itemstack:set_name("default:torch")
+	def.mesh = "torch_floor.obj"
+	def.selection_box = {
+		type = "wallmounted",
+		wall_bottom = {-1/8, -1/2, -1/8, 1/8, 2/16, 1/8}
+	}
+	minetest.register_node(":" .. name, def)
 
-		return itemstack
-	end,
-	floodable = true,
-	on_flood = on_flood,
-	on_rotate = false
-})
+	local torch_wall = table.copy(def)
+	torch_wall.mesh = "torch_wall.obj"
+	torch_wall.selection_box = {
+		type = "wallmounted",
+		wall_side = {-1/2, -1/2, -1/8, -1/8, 1/8, 1/8}
+	}
+	torch_wall.groups.not_in_creative_inventory = 1
 
-minetest.register_node("default:torch_wall", {
-	drawtype = "mesh",
-	mesh = "torch_wall.obj",
-	tiles = {{
-		    name = "default_torch_on_floor_animated.png",
-		    animation = {type = "vertical_frames", aspect_w = 16, aspect_h = 16, length = 3.3}
-	}},
-	paramtype = "light",
-	paramtype2 = "wallmounted",
-	sunlight_propagates = true,
-	walkable = false,
-	light_source = 12,
-	groups = {choppy=2, dig_immediate=3, flammable=1, not_in_creative_inventory=1, attached_node=1, torch=1},
-	drop = "default:torch",
-	selection_box = {
+	local torch_ceiling = table.copy(def)
+	torch_ceiling.mesh = "torch_ceiling.obj"
+	torch_ceiling.selection_box = {
 		type = "wallmounted",
-		wall_side = {-1/2, -1/2, -1/8, -1/8, 1/8, 1/8},
-	},
-	sounds = default.node_sound_wood_defaults(),
-	floodable = true,
-	on_flood = on_flood,
-	on_rotate = false
-})
+		wall_top = {-1/8, -1/16, -5/16, 1/8, 1/2, 1/8}
+	}
+	torch_ceiling.groups.not_in_creative_inventory = 1
+
+	minetest.register_node(":" .. name .. "_wall", torch_wall)
+	minetest.register_node(":" .. name .. "_ceiling", torch_ceiling)
+end
 
-minetest.register_node("default:torch_ceiling", {
-	drawtype = "mesh",
-	mesh = "torch_ceiling.obj",
+default.register_torch("default:torch", {
+	description = S("Torch"),
 	tiles = {{
-		    name = "default_torch_on_floor_animated.png",
-		    animation = {type = "vertical_frames", aspect_w = 16, aspect_h = 16, length = 3.3}
+		name = "default_torch_mesh_animated.png",
```

Comment:
is there a reason the names were changed?
It would be better to keep the old names so texture packs don't have to update.

Reply:
https://github.com/minetest/minetest_game/pull/2436#issuecomment-569530941